lib/modules/blockchain_process/blockchain.js

```diff
@@ -177,19 +183,33 @@ Blockchain.prototype.run = function() {
     self.child.stdout.on('data', (data) => {
       console.error(`Geth error: ${data}`);
     });
+    let httpReady = false;
+    let wsReady = !self.config.wsRPC;
     // Geth logs appear in stderr somehow
-    self.child.stderr.on('data', (data) => {
+    self.child.stderr.on('data', async (data) => {
       data = data.toString();
-      if (!self.readyCalled && data.indexOf('WebSocket endpoint opened') > -1) {
+      if (data.indexOf('HTTP endpoint opened') > -1) {
+        httpReady = true;
+      }
+
+      if (data.indexOf('WebSocket endpoint opened') > -1) {
+        wsReady = true;
+      }
+
+      if (!self.readyCalled && wsReady && httpReady) {
+        self.readyCalled = true;
```

Comment:
this is the first things to do in case we receive other log while we wait for the proxy to start =>
`await self.setupProxy()`

Reply:
The change you made makes sense, nice work!! Going back to my earlier comments, relying on both HTTP endpoints and WebSockets endpoints being available kind of limits our ability to use geth with just an HTTP endpoint or just a WebSockets endpoint.

Reply:
I haven't found the config for the just ws endpoint.
However we have 1 to say I don't want ws

This is why I initialise wsReady with:
```let wsReady = !self.config.wsRPC;```
That way if you specify that you don't need ws then it is already true

Is it what you are referring to?

Reply:
Yes, so if `wsRPC` is disabled, and therefore `wsReady` is never true, the proxy will never be initialised, and `self.readyCalled` will never be called?